tests/functional/conftest.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+import pytest
+import webtest
+
+from tests.simple_server import serve_content
+from viahtml.app import Application
+
+
+@pytest.fixture(scope="session")  # pragma: no cover
+def app(with_environ):
+    app = Application.create()
+    app.debug = True
+
+    return webtest.TestApp(app)
+
+
+@pytest.fixture(scope="session")  # pragma: no cover
+def with_environ():
+    # WSGI uses repeated elements to express this, which means we can't use
+    # the standard configparser.ConfigParser to read them. So they are
+    # duplicated here:
+    os.environ.update(
+        {
+            "VIA_H_EMBED_URL": "http://localhost:3001/hypothesis",
+            "VIA_IGNORE_PREFIXES": "http://localhost:5000/,http://localhost:3001/,https://localhost:5000/,https://localhost:3001/",
+            "VIA_DEBUG": "1",
+        }
+    )
```

Comment:
I'm not sure what the `with_` in the name is about?

```suggestion
@pytest.fixture(scope="session")  # pragma: no cover
def environ():
    # WSGI uses repeated elements to express this, which means we can't use
    # the standard configparser.ConfigParser to read them. So they are
    # duplicated here:
    os.environ.update(
        {
            "VIA_H_EMBED_URL": "http://localhost:3001/hypothesis",
            "VIA_IGNORE_PREFIXES": "http://localhost:5000/,http://localhost:3001/,https://localhost:5000/,https://localhost:3001/",
            "VIA_DEBUG": "1",
        }
    )
    return os.environ
```

Reply:
I think you've sort of spotted it and missed it at the same time. When we intend fixtures to be using only to establish a situation rather than to be used as a return value, we do two things to indicate that:

 * Verbify the name
 * Remove the return value

For example a fixture you intend to use the value of might be called 'feature_flag' but if it's only intended to create a situation it might be called 'with_feature_flag_on` and not return anything.

Usually we do `mark.use_fixtures` with these types of fixture, but in this case it didn't work. I could also just inline this into the app fixture and avoid the whole question?